src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -575,15 +575,29 @@ def process_metric_update_namespace(namespace):
 
 
 def datetime_string_type(string):
-    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M\'Z\''. '''
-    date_format = '%Y-%m-%dT%H:%MZ'
-    return datetime.strptime(string, date_format).strftime(date_format)
+    """ Validates UTC datettime in accepted format. Examples: 2017-12-31T01:11:59Z,
+    2017-12-31T01:11Z or 2017-12-31T01Z or 2017-12-31 """
+    accepted_date_formats = ['%Y-%m-%dT%H:%M:%SZ', '%Y-%m-%dT%H:%MZ',
+                             '%Y-%m-%dT%HZ', '%Y-%m-%d']
+    for form in accepted_date_formats:
+        try:
+            return datetime.strptime(string, form).strftime(form)
```

Comment:
It seems like this could be refactored to leverage the existing datetime_type method and avoid duplicating the logic here. You could merge these into one method with a parameter to toggle the behavior:

```Python
get_datetime_type_method(to_string=False):
   def datetime_type(string):
      # stuff goes here
      if to_string:
        return datetime.strptime(string, form).strftime(form)
     else:
        return datetime.strptime(string, form)
   return datetime_type
```